modules/src/ZeroTier/api/module.php

```diff
@@ -0,0 +1,177 @@
+<?php namespace pineapple;
+// 2023 - m5kro
+
+class ZeroTier extends Module
+{
+
+    public function route()
+    {
+        switch ($this->request->action) {
+
+            case "getDependenciesStatus":
+                $this->getDependenciesStatus();
+                break;
+
+            case "managerDependencies":
+                $this->managerDependencies();
+                break;
+
+            case "getDependenciesInstallStatus":
+                $this->getDependenciesInstallStatus();
+                break;
+            
+            case "getZeroTierStatus":
+                $this->getZeroTierStatus();
+                break;
+
+            case "zerotierSwitch":
+                $this->zerotierSwitch();
+                break;
+
+            case "zerotierBootSwitch":
+                $this->zerotierBootSwitch();
+                break;
+
+            case "zerotierSetID":
+                $this->zerotierSetID();
+                break;
+
+            case "zerotierGetID":
+                $this->zerotierGetID();
+                break;
+
+            case "zerotierNewIdentity":
+                $this->zerotierNewIdentity();
+                break;
+
+            case "zerotierGetIdentity":
+                $this->zerotierGetIdentity();
+                break;
+        }
+    }
+
+    // Thanks to xchwarze (DSR) for most of the dependencies code
+    protected function getDependenciesStatus()
+    {
+        $response = [
+            "installed" => false,
+            "install" => "Install",
+            "installLabel" => "success",
+            "processing" => false,
+        ];
+
+        if (file_exists("/tmp/zerotier.progress")) {
+            $response["install"] = "Installing...";
+            $response["installLabel"] = "warning";
+            $response["processing"] = true;
+        
+        } else if ($this->checkDependencyInstalled()) {
+            $response["install"] = "Remove";
+            $response["installLabel"] = "danger";
+            $response["installed"] = true;
+        }
+
+        $this->response = $response;
+    }
+
+    protected function checkDependencyInstalled()
+    {      
+        if ($this->checkDependency("zerotier-cli")) {
+            return true; 
+        }
+
+        return false;
+    }
+
+    protected function managerDependencies()
+    {
+        $action = $this->checkDependencyInstalled() ? "remove" : "install";
+        $command = "bash /pineapple/modules/ZeroTier/scripts/dependencies.sh";
```

Comment:
Wouldn't the execBackground() calls already be executing bash?
I don't remember having to add that in the modules I made.

Reply:
I added bash because this module was not originally built to be installed with the module manager but though copy and paste. This meant that `chmod +x` was not applied to dependencies.sh so I had to add bash in front.

It has been removed.